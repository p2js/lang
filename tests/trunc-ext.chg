extern "intrinsic" fn trunc<T, U>(v: T): U;
extern "intrinsic" fn ext<T, U>(v: T): U;

fn main() {
    trunc::<i8, u8>(0i8);
    trunc::<i8, u16>(0i8);
    trunc::<i8, u64>(0i8);
    trunc::<u64, i8>(0u64);
    trunc::<u64, u8>(0u64);
    
    ext::<i8, u8>(0i8);
    ext::<i8, u16>(0i8);
    ext::<u8, u16>(0u8);
    ext::<u8, i16>(0u8);
    ext::<i8, u64>(0i8);
    ext::<u64, i8>(0u64);
    ext::<u64, u8>(0u64);
}
