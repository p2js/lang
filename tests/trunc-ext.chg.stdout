expr @ 5:19 5:20 => extern "intrinsic" fn trunc(...)
expr @ 5:21 5:22 => i8
expr @ 5:19 5:25 => u8
expr @ 6:20 6:21 => extern "intrinsic" fn trunc(...)
expr @ 6:22 6:23 => i8
expr @ 6:20 6:26 => u16
expr @ 7:20 7:21 => extern "intrinsic" fn trunc(...)
expr @ 7:22 7:23 => i8
expr @ 7:20 7:26 => u64
expr @ 8:20 8:21 => extern "intrinsic" fn trunc(...)
expr @ 8:22 8:23 => u64
expr @ 8:20 8:27 => i8
expr @ 9:20 9:21 => extern "intrinsic" fn trunc(...)
expr @ 9:22 9:23 => u64
expr @ 9:20 9:27 => u8
expr @ 11:17 11:18 => extern "intrinsic" fn ext(...)
expr @ 11:19 11:20 => i8
expr @ 11:17 11:23 => u8
expr @ 12:18 12:19 => extern "intrinsic" fn ext(...)
expr @ 12:20 12:21 => i8
expr @ 12:18 12:24 => u16
expr @ 13:18 13:19 => extern "intrinsic" fn ext(...)
expr @ 13:20 13:21 => u8
expr @ 13:18 13:24 => u16
expr @ 14:18 14:19 => extern "intrinsic" fn ext(...)
expr @ 14:20 14:21 => u8
expr @ 14:18 14:24 => i16
expr @ 15:18 15:19 => extern "intrinsic" fn ext(...)
expr @ 15:20 15:21 => i8
expr @ 15:18 15:24 => u64
expr @ 16:18 16:19 => extern "intrinsic" fn ext(...)
expr @ 16:20 16:21 => u64
expr @ 16:18 16:25 => i8
expr @ 17:18 17:19 => extern "intrinsic" fn ext(...)
expr @ 17:20 17:21 => u64
expr @ 17:18 17:25 => u8
expr @ 4:11 19:1 => ()



define void @main() {
prologue:
br label %bb.0

bb.0:
%l.0 = trunc i8 0 to i8
%l.1 = trunc i8 0 to i16
%l.2 = trunc i8 0 to i64
%l.3 = trunc i64 0 to i8
%l.4 = trunc i64 0 to i8
%l.5 = sext i8 0 to i8
%l.6 = sext i8 0 to i16
%l.7 = zext i8 0 to i16
%l.8 = zext i8 0 to i16
%l.9 = sext i8 0 to i64
%l.10 = zext i64 0 to i8
%l.11 = zext i64 0 to i8
ret void

}


